src/main/webapp/app/modeling-submission/modeling-submission.component.ts

```diff
@@ -224,106 +229,73 @@ export class ModelingSubmissionComponent implements OnInit, OnDestroy, Component
-                    this.submission = submission.body!;
-                    this.result = this.submission.result;
-                    this.isSaving = false;
-                    this.jhiAlertService.success('artemisApp.modelingEditor.saveSuccessful');
-                    this.isActive = this.modelingExercise.dueDate == null || new Date() <= moment(this.modelingExercise.dueDate).toDate();
-                    this.subscribeToAutomaticSubmissionWebsocket();
-                },
-                error => {
-                    this.jhiAlertService.error('artemisApp.modelingEditor.error');
-                    this.isSaving = false;
-                },
-            );
-        }
-    }
-
-    submit(): void {
-        if (!this.submission) {
-            return;
-        }
+        this.submission.submitted = true;
         this.updateSubmissionModel();
         if (this.isModelEmpty(this.submission.model)) {
             this.jhiAlertService.warning('artemisApp.modelingEditor.empty');
             return;
         }
-
-        let confirmSubmit = true;
-        if (this.calculateNumberOfModelElements() < 10) {
-            confirmSubmit = window.confirm('Are you sure you want to submit? You cannot edit your model anymore until you get an assessment!');
-        }
-
-        if (confirmSubmit) {
-            this.submission.submitted = true;
+        this.isSaving = true;
+        this.autoSaveTimer = 0;
+        if (this.submission.id) {
             this.modelingSubmissionService.update(this.submission, this.modelingExercise.id).subscribe(
                 response => {
                     this.submission = response.body!;
                     this.umlModel = JSON.parse(this.submission.model);
                     this.result = this.submission.result;
                     this.retryStarted = false;
 
-                    if (this.isActive) {
-                        this.jhiAlertService.success('artemisApp.modelingEditor.submitSuccessful');
-                    } else {
+                    if (this.isLate) {
                         this.jhiAlertService.warning('artemisApp.modelingEditor.submitDeadlineMissed');
+                    } else {
+                        this.jhiAlertService.success('artemisApp.modelingEditor.submitSuccessful');
                     }
 
                     this.subscribeToWebsockets();
                     if (this.automaticSubmissionWebsocketChannel) {
                         this.jhiWebsocketService.unsubscribe(this.automaticSubmissionWebsocketChannel);
                     }
                 },
-                err => {
+                () => {
                     this.jhiAlertService.error('artemisApp.modelingEditor.error');
                     this.submission.submitted = false;
                 },
+                () => (this.isSaving = false),
+            );
+        } else {
+            this.modelingSubmissionService.create(this.submission, this.modelingExercise.id).subscribe(
```

Comment:
why did you delete update here?
I think we should always use update instead of create

Reply:
I didn't delete it. I only check if we already have submission, then we update it, if not then create.